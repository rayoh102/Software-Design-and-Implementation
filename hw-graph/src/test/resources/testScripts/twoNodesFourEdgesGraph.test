# Test adding two nodes, with two edges between the two nodes and two edges on the same nodes

# Create a graph
CreateGraph twoNodesFourEdgesGraph

# Add two nodes
AddNode twoNodesFourEdgesGraph a
AddNode twoNodesFourEdgesGraph b

# Add four edges
AddEdge twoNodesFourEdgesGraph a a AA
AddEdge twoNodesFourEdgesGraph a b AB
AddEdge twoNodesFourEdgesGraph b a BA
AddEdge twoNodesFourEdgesGraph b b BB

# Print the nodes in the graph, the outgoing edges from a and the outgoing edges from b
ListNodestwoNodesFourEdgesGraph
ListChildren twoNodesFourEdgesGraph a
ListChildren twoNodesFourEdgesGraph b