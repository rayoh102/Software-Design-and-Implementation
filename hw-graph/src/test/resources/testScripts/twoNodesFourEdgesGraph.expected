# Test adding two nodes, with two edges between the two nodes and two edges on the same nodes

# Create a graph
created graph twoNodesFourEdgesGraph


# Add two nodes
added node a to twoNodesFourEdgesGraph
added node b to twoNodesFourEdgesGraph


# Add four edges
added edge AA from a to a in twoNodesFourEdgesGraph
added edge AB from a to b in twoNodesFourEdgesGraph
added edge BA from b to a in twoNodesFourEdgesGraph
added edge BB from b to b in twoNodesFourEdgesGraph


# Print the nodes in the graph, the outgoing edges from a and the outgoing edges from b
the children of a in twoNodesFourEdgesGraph are: a(AA) b(AB)
the children of b in twoNodesFourEdgesGraph are: a(BA) b(BB)